### PR TITLE
XENOPS-794 Replaced Jenkinsfile with GitHub workflow

### PR DESCRIPTION
**Pull request by todorinskiz**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        run: ./gradlew check
+#      - name: Upload analysis to sonarcloud
+#        env:
+#          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+#          GITHUB_TOKEN: ${{ github.token }}
+#        run: ./gradlew sonarqube -Dsonar.projectKey=xenit-eu_alfresco-gradle-sdk -Dsonar.organization=xenit-eu -Dsonar.host.url=https://sonarcloud.io
+      - name: Publish
+        if: ${{ github.ref == 'refs/heads/master' || startswith(github.ref, 'refs/tags/') }}
+        env:
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.MAVEN_CENTRAL_GPG_KEY }}
```
Comment (vierbergenlars) [inline .github/workflows/ci.yml:24]: To use this signing key, you need to use in-memory signing: https://docs.gradle.org/current/userguide/signing_plugin.html#sec:in-memory-keys

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.MAVEN_CENTRAL_GPG_PASSWORD }}
+          ORG_GRADLE_PROJECT_sonatype_username: ${{ secrets.MAVEN_CENTRAL_USERNAME }}
+          ORG_GRADLE_PROJECT_sonatype_password: ${{ secrets.MAVEN_CENTRAL_PASSWORD }}
+        run: ./gradlew publish -PsigningKeyId=CDE3528F
```
Comment (vierbergenlars) [inline .github/workflows/ci.yml:28]: Remove the signing key ID, we don't need it when using in-memory signing.

File: alfresco-zipkin-repo/build.gradle [hunk 3]
```diff
@@ -1,6 +1,8 @@
 plugins {
     id 'eu.xenit.alfresco' version "0.2.0"
     id "eu.xenit.amp" version "0.2.0"
+    id "idea"
```
Comment (vierbergenlars) [inline alfresco-zipkin-repo/build.gradle:4]: Don't apply the `idea` plugin to your projects

File: alfresco-zipkin-share/build.gradle [hunk 4]
```diff
@@ -1,6 +1,8 @@
 plugins {
     id 'eu.xenit.alfresco' version "0.2.0"
     id "eu.xenit.amp" version "0.2.0"
+    id "idea"
```
Comment (vierbergenlars) [inline alfresco-zipkin-share/build.gradle:4]: Don't apply the `idea` plugin to your projects

File: build.gradle [hunk 5]
```diff
@@ -1,17 +1,33 @@
...
+    project.ext {
+        alfresco_version = '5.2.f'
+        brave_version = '5.13.3'
+        zipkin_reporter_version = '2.16.3'
+    }
+
+    pluginManager.withPlugin('java') {
+        repositories {
+            mavenCentral()
+            jcenter()
```
Comment (vierbergenlars) [inline build.gradle:23]: Remove jcenter if possible.

File: integration-tests/build.gradle [hunk 6]
```diff
@@ -1,5 +1,7 @@
 plugins {
     id "eu.xenit.docker-compose"
+    id "java"
+    id "idea"
```
Comment (vierbergenlars) [inline integration-tests/build.gradle:4]: Don't apply the `idea` plugin

**Review by vierbergenlars — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (todorinskiz) [reply .github/workflows/ci.yml:24]: done in latest commit

File: .github/workflows/ci.yml (hunk 2 above)
Comment (todorinskiz) [reply .github/workflows/ci.yml]: done in latest commit

File: alfresco-zipkin-repo/build.gradle (hunk 3 above)
Comment (todorinskiz) [reply alfresco-zipkin-repo/build.gradle]: done in latest commit

File: alfresco-zipkin-share/build.gradle (hunk 4 above)
Comment (todorinskiz) [reply alfresco-zipkin-share/build.gradle]: done in latest commit

File: build.gradle (hunk 5 above)
Comment (todorinskiz) [reply build.gradle]: done in latest commit

File: integration-tests/build.gradle (hunk 6 above)
Comment (todorinskiz) [reply integration-tests/build.gradle]: done in latest commit

**Review by vierbergenlars — APPROVED** [review]

**Pull request merged**